Zig.gitignore

```diff
@@ -0,0 +1,4 @@
+# This file is for zig-specific build artifacts.
+
+zig-cache/
```

Comment:
As per the recent 0.13.0 release of Zig, `zig-cache/` has been renamed to `.zig-cache/`. See https://ziglang.org/download/0.13.0/release-notes.html#codezig-cachecode-renamed-to-codezig-cachecode.

I've opened a PR that proposes an up-to-date `Zig.gitignore`: https://github.com/github/gitignore/pull/4503